image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml

```diff
@@ -233,4 +239,10 @@ spec:
       - name: {{ $extraMount.name }}
         {{- $extraMount.source | toYaml | nindent 8 }}
       {{- end }}
+      {{- range $mount := (default list ._rox.central.declarativeConfiguration.mounts) }}
+      - name: {{ $mount }}
+        configMap:
+          name: {{ $mount }}
+          optional: true
```

Comment:
Why do we make this mount optional - do we expect it to be deleted? 

What will happen if the resource contained in the mount is upserted within Postgres, but after the restart, the mount is not existing anymore? Would the resource be deleted from Postgres? 

Also, what if a declarative resource from one mount references a declarative resource from another mount?

Reply:
>Why do we make this mount optional - do we expect it to be deleted?

The reason to make this mount optional is that, in theory the config map could be deleted / misconfigured etc., but most importantly, we do not want central to _not_ come up. If we wouldn't set `optional==true`, the central pod wouldn't come up without the config map being there. Since declarative config isn't _required_ for a central to become "ready", I've decided to make it optional (we use this pattern across all mounts that are not required for central startup as well, I do like to be quite conservative here about what is required for a central boot).

 >What will happen if the resource contained in the mount is upserted within Postgres, but after the restart, the mount is not existing anymore? Would the resource be deleted from Postgres?

I think these questions are more or less jumping ahead in terms of the context of this PR and are rather part of [ROX-14147](https://issues.redhat.com/browse/ROX-14147). Without looking at the actual implementation, the expectation would be exactly what you stated: in case the declarative config has been removed, the resources should be removed as well (simply put, we want a reconciliation on the contents).

>Also, what if a declarative resource from one mount references a declarative resource from another mount?

I think that's an interesting question, and definitely a possibility. I suppose, again, this should be part of [ROX-14147](https://issues.redhat.com/browse/ROX-14147). The reconciliation should first read all contents of the mounts, and then "transform" the configuration to proto resources. There we should take care of resolving these references.

Note that the references part is still somewhat missing in JIRA, I'll be adding this shortly. Those are definitely cases we have to cover.